src/component/AddBookForm.js

```diff
@@ -1,17 +1,40 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/books';
 import './AddBookForm.css';
 
 function AddBookForm() {
+  let inputName;
+  let inputCategory;
+
+  const dispatch = useDispatch();
+
   return (
     <section id="form-section">
       <h2 className={AddBookForm.desc}>Add new book</h2>
-      <form id="add-book-form">
-        <input type="text" name="title" placeholder="Book title" required />
-        <select>
-          <option value="">Category</option>
-          <option value="saab">Business</option>
-          <option value="opel">Leadership</option>
-          <option value="audi">History</option>
+      <form
+        id="add-book-form"
+        onSubmit={(e) => {
+          e.preventDefault();
+          if (!inputName.value.trim() || !inputCategory.value.trim()) {
+            return;
+          }
+          const book = {
+            name: inputName.value,
+            category: inputCategory.value,
+            author: 'Abenezer',
+          };
```

Comment:
- [ ] Please generate a unique `id` for each book object. You can take a look at [uuid4](https://www.npmjs.com/package/uuid4).

Reply:
Ok I will do that .